cpp/src/arrow/acero/query_context.h

```diff
@@ -145,11 +144,6 @@ class ARROW_ACERO_EXPORT QueryContext {
   std::unique_ptr<TaskScheduler> task_scheduler_ = TaskScheduler::Make();
 
   ThreadIndexer thread_indexer_;
-  struct ThreadLocalData {
-    bool is_init = false;
-    arrow::util::TempVectorStack stack;
```

Comment:
It seems this was the last place in a non-internal header file where the full definition of `TempVectorStack` is required, so, perhaps we can now move `TempVectorStack` to a `arrow/compute/util_internal.h` header file?

Reply:
Very thoughtful suggestion. Will do.

Reply:
Addressed. And reduced some more (maybe unrelated) includings.